src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts

```diff
@@ -331,63 +452,59 @@ g.test('color_textures,non_compressed,array')
-          // destination texture
-          ...F.kCopyBoxOffsetsForWholeDepth,
-
-          // Copy 1 texture slice from the 1st slice of the source texture to the 1st slice of the
-          // destination texture.
-          {
-            srcOffset: { x: 0, y: 0, z: 0 },
-            dstOffset: { x: 0, y: 0, z: 0 },
-            copyExtent: { width: 0, height: 0, depth: -2 },
-          },
-          // Copy 1 texture slice from the 2nd slice of the source texture to the 2nd slice of the
-          // destination texture.
-          {
-            srcOffset: { x: 0, y: 0, z: 1 },
-            dstOffset: { x: 0, y: 0, z: 1 },
-            copyExtent: { width: 0, height: 0, depth: -3 },
-          },
-          // Copy 1 texture slice from the 1st slice of the source texture to the 2nd slice of the
-          // destination texture.
-          {
-            srcOffset: { x: 0, y: 0, z: 0 },
-            dstOffset: { x: 0, y: 0, z: 1 },
-            copyExtent: { width: 0, height: 0, depth: -1 },
-          },
-          // Copy 1 texture slice from the 2nd slice of the source texture to the 1st slice of the
-          // destination texture.
-          {
-            srcOffset: { x: 0, y: 0, z: 1 },
-            dstOffset: { x: 0, y: 0, z: 0 },
-            copyExtent: { width: 0, height: 0, depth: -1 },
-          },
-          // Copy 2 texture slices from the 1st slice of the source texture to the 1st slice of the
-          // destination texture.
+        poptions('textureSize', [
+          // The heights and widths are all power of 2
           {
-            srcOffset: { x: 0, y: 0, z: 0 },
-            dstOffset: { x: 0, y: 0, z: 0 },
-            copyExtent: { width: 0, height: 0, depth: -3 },
+            srcTextureSize: { width: 8, height: 8, depth: 5 },
+            dstTextureSize: { width: 8, height: 8, depth: 5 },
           },
-          // Copy 3 texture slices from the 2nd slice of the source texture to the 2nd slice of the
-          // destination texture.
+          // None of the widths or heights are power of 2
           {
-            srcOffset: { x: 0, y: 0, z: 1 },
-            dstOffset: { x: 0, y: 0, z: 1 },
-            copyExtent: { width: 0, height: 0, depth: -1 },
+            srcTextureSize: { width: 60, height: 52, depth: 5 },
+            dstTextureSize: { width: 60, height: 52, depth: 5 },
           },
         ])
       )
-      .combine(poptions('srcCopyLevel', [0, 3]))
-      .combine(poptions('dstCopyLevel', [0, 3]))
+      .combine(poptions('format', kCompressedTextureFormats))
+      .combine(poptions('copyBoxOffsets', F.kCopyBoxOffsetsFor2DArrayTextures))
+      .combine(poptions('srcCopyLevel', [0, 2]))
+      .combine(poptions('dstCopyLevel', [0, 2]))
   )
   .fn(async t => {
     const { textureSize, format, copyBoxOffsets, srcCopyLevel, dstCopyLevel } = t.params;
 
+    const extension: GPUExtensionName | undefined = kAllTextureFormatInfo[format].extension;
```

Comment:
nit: If you use kCompressedTextureFormatInfo here, then you shouldn't need the assert (I think)